### ISSUE TITLE
 install calico init container install-cni failed -dial tcp 10.96.0.1:443: i/o timeout

### ISSUE THREAD

**Issue by DemoRoyalD** [projectcalico/calico#8463]
# calico 3.26.1 k8s v1.28.2 containerruntime containerd

## Expected Behavior
excepte calico-node normal running ,but it now crash 

## Current Behavior

install calico : kubectl apply -f https://raw.githubusercontent.com/projectcalico/calico/v3.26.1/manifests/calico.yaml 
 kubectl get po -nkube-system -owide 
```text
calico-kube-controllers-6c9bb74768-cn99s      0/1     Terminating             0              62m     <none>         10.10.101.45-worker   <none>           <none>
calico-kube-controllers-6c9bb74768-td8wb      0/1     ContainerCreating       0              54m     <none>         10.10.101.45-worker   <none>           <none>
calico-node-784sk                             0/1     Init:CrashLoopBackOff   4 (32s ago)    6m13s   10.10.101.44   10.10.101.44-worker   <none>           <none>
calico-node-gfmfd                             0/1     Init:CrashLoopBackOff   4 (45s ago)    6m13s   10.10.101.45   10.10.101.45-worker   <none>           <none>
calico-node-tfk7t                             1/1     Running                 0              54m     10.10.101.43   10.10.101.43-master   <none>           <none>
coredns-66f779496c-89vm2                      0/1     ContainerCreating       0              7h7m    <none>         10.10.101.44-worker   <none>           <none>
coredns-66f779496c-x7ssh                      0/1     ContainerCreating       0              7h7m    <none>         10.10.101.45-worker   <none>           <none>
etcd-10.10.101.43-master                      1/1     Running                 0              26h     10.10.101.43   10.10.101.43-master   <none>           <none>
kube-apiserver-10.10.101.43-master            1/1     Running                 1 (174m ago)   150m    10.10.101.43   10.10.101.43-master   <none>           <none>
kube-controller-manager-10.10.101.43-master   1/1     Running                 14 (52m ago)   26h     10.10.101.43   10.10.101.43-master   <none>           <none>
kube-proxy-6zvx2                              1/1     Running                 0              25h     10.10.101.45   10.10.101.45-worker   <none>           <none>
kube-proxy-h5sh9                              1/1     Running                 0              25h     10.10.101.44   10.10.101.44-worker   <none>           <none>
kube-proxy-qfhqs                              1/1     Running                 0              26h     10.10.101.43   10.10.101.43-master   <none>           <none>
kube-scheduler-10.10.101.43-master            1/1     Running                 14 (52m ago)   26h     10.10.101.43   10.10.101.43-master   <none>           <none>
```

find coredns can't work ,beacause of calico plugins can't install 

so i install colico ,and find calico-node running in master node , but can't in worker node 

kubectl logs -f calico-node-784sk  -nkube-system -c install-cni  

# kubectl logs -f calico-node-784sk  -nkube-system -c install-cni 
2024-01-30 08:47:35.973 [INFO][1] cni-installer/<nil> <nil>: Running as a Kubernetes pod
2024-01-30 08:47:47.388 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/bandwidth"
2024-01-30 08:47:47.388 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/bandwidth
2024-01-30 08:47:47.643 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/calico"
2024-01-30 08:47:47.643 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/calico
2024-01-30 08:47:47.878 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/calico-ipam"
2024-01-30 08:47:47.878 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/calico-ipam
2024-01-30 08:47:47.886 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/flannel"
2024-01-30 08:47:47.886 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/flannel
2024-01-30 08:47:47.901 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/host-local"
2024-01-30 08:47:47.901 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/host-local
2024-01-30 08:47:48.152 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/install"
2024-01-30 08:47:48.152 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/install
2024-01-30 08:47:48.165 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/loopback"
2024-01-30 08:47:48.165 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/loopback
2024-01-30 08:47:48.183 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/portmap"
2024-01-30 08:47:48.183 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/portmap
2024-01-30 08:47:48.198 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/tuning"
2024-01-30 08:47:48.199 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/tuning
2024-01-30 08:47:48.199 [INFO][1] cni-installer/<nil> <nil>: Wrote Calico CNI binaries to /host/opt/cni/bin

2024-01-30 08:47:48.285 [INFO][1] cni-installer/<nil> <nil>: CNI plugin version: v3.26.1

2024-01-30 08:47:48.285 [INFO][1] cni-installer/<nil> <nil>: /host/secondary-bin-dir is not writeable, skipping
W0130 08:47:48.285528       1 client_config.go:618] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
2024-01-30 08:48:18.320 [ERROR][1] cni-installer/<nil> <nil>: Unable to create token for CNI kubeconfig error=Post "https://10.96.0.1:443/api/v1/namespaces/kube-system/serviceaccounts/calico-cni-plugin/token": dial tcp 10.96.0.1:443: i/o timeout
2024-01-30 08:48:18.320 [FATAL][1] cni-installer/<nil> <nil>: Unable to create token for CNI kubeconfig error=Post "https://10.96.0.1:443/api/v1/namespaces/kube-system/serviceaccounts/calico-cni-plugin/token": dial tcp 10.96.0.1:443: i/o timeout

## net route 
route is normal
# ip r 
default via 10.10.101.1 dev ens192 proto static metric 100 
10.10.101.0/24 dev ens192 proto kernel scope link src 10.10.101.43 metric 100 
64.64.64.0/26 dev docker0 proto kernel scope link src 64.64.64.1 
blackhole 192.168.195.0/26 proto bird

Has anyone encountered it，please help me

**Comment by DemoRoyalD** [comment]
and ping 10.96.0.1 is accessible in worker node ,but curl -k "[https://10.96.0.1:443/api/v1/namespaces/kube-system/serviceaccounts/calico-cni-plugin/token](https://10.96.0.1/api/v1/namespaces/kube-system/serviceaccounts/calico-cni-plugin/token)" no response

**Comment by DemoRoyalD** [comment]
and ping 10.96.0.1 is accessible in worker node ,but curl -k "[https://10.96.0.1:443/api/v1/namespaces/kube-system/serviceaccounts/calico-cni-plugin/token](https://10.96.0.1/api/v1/namespaces/kube-system/serviceaccounts/calico-cni-plugin/token)" no response ,it is block

**Comment by jon-nfc** [comment]
same issue, even deleting everything, including the host mount directories, rebooting, flushing firewall does not fix the problem.

**Comment by caseydavenport** [comment]
Please share a bit more about your environement:

- What cloud are you using?
- Are you using kube-proxy in iptables mode, IPVS mode, or nftables mode?

Most likely this is something to do with kube-proxy, or with access to the control plane nodes.

> and ping 10.96.0.1 is accessible in worker node

This is surprising - I wouldn't expect `ping` to work on a Service cluster IP. Do you mean `curl` works?

What about something like this:

```
curl -k https://10.96.0.1/api/v1/namespaces
```

**Comment by jon-nfc** [comment]
@caseydavenport, thanks for the `curl` command (had skipped my mind to use) it assisted in finding the cause in my case.

Setup: K3s, 1x Master, 1x Slave. 2x different providers. hostinger and oracle cloud.

Turns out my issue was the hosting providers firewall. 

The node that had the connectivity issue to the api was the master node, which was odd. Stopping the k3s service and running the binary manually `k3s server` (outputs log to stdout) yielded an error `cant connect to proxy` with the address being used was the master nodes external IP. From the master node `curl -k https://<internal IP>:6443/api/v1/namespaces` worked as did `curl -k https://<local lan ip>:6443/api/v1/namespaces`, however `curl -k https://<external IP>:6443/api/v1/namespaces` didn't. The slave node had no issues connecting.

The issue was self-inflicted as I had adjusted the hosting providers firewall from all hosts `0.0.0.0/0 TCP/6443` to `cluster nodes TCP/6443`. As the master node's external IP is nat'd, adding an allow rule for the external IP corrected access to the API from the master node.  cluster now working again
